spark/src/main/java/org/apache/iceberg/spark/SparkDataFile.java

```diff
@@ -33,37 +34,67 @@
 
 public class SparkDataFile implements DataFile {
 
+  private final int filePathPosition;
+  private final int fileFormatPosition;
+  private final int partitionPosition;
```

Comment:
This is a debatable decision. It felt safer not to rely on the order of the field in `DataFile` in case if something changes.

Reply:
I like this.